pkg/daemon/metrics.go

```diff
@@ -186,15 +186,18 @@ func extractRegularMetrics(processName, output string) (offsetFromMaster, maxOff
 	return
 }
 
-func addFlagsForMonitor(nodeProfile *ptpv1.PtpProfile) {
+func addFlagsForMonitor(nodeProfile *ptpv1.PtpProfile, stdoutToSocket bool) {
 	// If output doesn't exist we add it for the prometheus exporter
 	if nodeProfile.Phc2sysOpts != nil {
 		if !strings.Contains(*nodeProfile.Phc2sysOpts, "-m") {
 			glog.Info("adding -m to print messages to stdout for phc2sys to use prometheus exporter")
 			*nodeProfile.Phc2sysOpts = fmt.Sprintf("%s -m", *nodeProfile.Phc2sysOpts)
 		}
-
-		if !strings.Contains(*nodeProfile.Phc2sysOpts, "-u") {
+		// stdoutTOSocket is for side car to consume events, -u  will not generate logs with offset and clock state.
+		// disable -u for  events
+		if stdoutToSocket && strings.Contains(*nodeProfile.Phc2sysOpts, "-u"){
+				glog.Error("-u option will not generate clock state events,  remove -u option")
```

Comment:
will the sidecard work as expected?
if not we may need to abort starting the process it will be super hard to find this line in the linuxptp logs

Reply:
sidecar won't have this log information and it will ignore generating events. but even better I can catch this information in sidecar by looking at the log for  `rms`  text and print an error or info.